tests/rest/client/v2_alpha/test_sync.py

```diff
@@ -314,6 +315,144 @@ def test_sync_backwards_typing(self):
+            )
+        )
+
+        self.room_state = {
+            # We need to set the room's join rules to allow knocking
+            EventTypes.JoinRules: {
+                "content": {"join_rule": "xyz.amorgan.knock"},
+                "state_key": "",
+            },
+            # The rest are state events that are recommended to strip and send to clients
+            EventTypes.Name: {"content": {"name": "A cool room"}, "state_key": ""},
+            EventTypes.RoomAvatar: {
+                "content": {
+                    "info": {
+                        "h": 398,
+                        "mimetype": "image/jpeg",
+                        "size": 31037,
+                        "w": 394,
+                    },
+                    "url": "mxc://example.org/JWEIFJgwEIhweiWJE",
+                },
+                "state_key": "",
+            },
+            EventTypes.RoomEncryption: {
+                "content": {"algorithm": "m.megolm.v1.aes-sha2"},
+                "state_key": "",
+            },
+            EventTypes.CanonicalAlias: {
+                "content": {"alias": canonical_alias, "alt_aliases": []},
+                "state_key": "",
+            },
+        }
+
+        for event_type, event in self.room_state.items():
+            self.helper.send_state(
+                self.room_id, event_type, event["content"], tok=self.tok,
+            )
+
+        # We expect the knock event to be included as well
+        self.room_state[EventTypes.Member] = {
+            "content": {"membership": "xyz.amorgan.knock", "displayname": "knocker"},
+            "state_key": "@knocker:test",
+        }
+
+    def test_knock_room_state(self):
+        """Tests that /sync returns state from a room after knocking on it."""
+        # Knock on a room
+        request, channel = self.make_request(
+            "POST",
+            "/_matrix/client/unstable/xyz.amorgan.knock/%s" % (self.room_id,),
+            b"{}",
+            self.knocker_tok,
+            shorthand=False,
+        )
+        self.assertEquals(200, channel.code, channel.result)
+
+        # Check that /sync includes stripped state from the room
+        request, channel = self.make_request(
+            "GET", self.url % self.next_batch, access_token=self.knocker_tok,
+        )
+        self.assertEqual(channel.code, 200, channel.json_body)
+
+        knock_entry = channel.json_body["rooms"]["xyz.amorgan.knock"]
+        room_state_events = knock_entry[self.room_id]["knock_state"]["events"]
+        for event in room_state_events:
+            event_type = event["type"]
+            if event_type not in self.room_state:
+                raise Exception(
+                    "Unexpected room state type '%s' in knock sync result"
+                    % (event_type,)
+                )
+
+            # Check the state content matches
+            self.assertDictEqual(
+                self.room_state[event_type]["content"], event["content"]
+            )
```

Comment:
Why not just `assertEquals`? From the [python docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertDictEqual):

> Test that two dictionaries are equal. If not, an error message is constructed that shows the differences in the dictionaries. This method will be used by default to compare dictionaries in calls to `assertEqual()`.

Reply:
Oh, TIL!